app/models/ext_management_system.rb

```diff
@@ -426,6 +426,32 @@ def enable!
     update!(:enabled => true)
   end
 
+  # override destroy_queue from AsyncDeleteMixin
+  def destroy_queue(deliver_on = Time.now)
+    _log.info("Queuing destroy of #{self.class.name} with id: #{id}")
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "orchestrate_destroy",
+      :deliver_on  => deliver_on
+    )
+  end
+
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    if self.destroy == false
+      raise MiqException::MiqQueueRetryLater.new(:deliver_on => 15.seconds.from_now)
```

Comment:
If you skip `deliver_on` whats the default? 
And is there a throttling in place?
Will the `retry` be until it succeeds or does it give up eventually?